### PR TITLE
Fix various typings and issues with gsap and greensock

### PR DESCRIPTION
**Pull request by ProbablePrime**
I found a number of discrepancies and issues when trying to use the gsap(https://greensock.com/) node module with TypeScript.

- The typings for `gsap` appeared to be incomplete and only supplied `TweenLite`
- There were duplicate typings in `greensock` this is [not an npm module](https://www.npmjs.com/package/greensock) and I therefore consider it to be a completely invalid folder. However its typings were better than `gsap`

To fix this i've merged `gsap` with `greensock` and added comments as appropriate. I then setup gsap to be exported in a similar pattern to [rx](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/rx/index.d.ts)

This should also address: #14193 , #13400, #11265

- [X] Make your PR against the `master` branch.
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] Run `tsc` without errors.
- [X] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://greensock.com/docs/#/HTML5/All
- [x] Increase the version number in the header if appropriate.
- [X] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

I'm going to tag @lucap86  as they have an open PR as well as: @codeBelt & @vilic the authors of the original typings for `gsap` and `greensock`

### CODE REVIEW

**Comment by ProbablePrime** [comment]
Regarding the build failure, any advice on typing these?: https://greensock.com/docs/#/HTML5/All/Power0/

They're aliases for naming reasons.

File: gsap/Ease.d.ts [hunk 1]
```diff
@@ -1,6 +1,113 @@
...
+    }
+
+    export interface RoughEaseConfig {
+        clamp?: boolean;
+        points?: number;
+        randomize?: boolean;
+        strength?: number;
+        taper?: 'in' | 'out' | 'both' | 'none';
+        template?: Ease;
+    };
```
Comment (DanielRosenwasser) [inline gsap/Ease.d.ts]: Remove the semicolon to get rid of the build failure

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

File: gsap/Ease.d.ts (hunk 1 above)
Comment (ProbablePrime) [inline gsap/Ease.d.ts]: Good catch thanks!

**Review by DanielRosenwasser — APPROVED** [review]

File: gsap/Animation.d.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        totalDuration(): number;
+        totalDuration(value: number): Animation;
+
+        /** Gets or sets the animation's total progress which is a value between 0 and 1 indicating the position of the virtual playhead (including repeats) where 0 is at the beginning, 0.5 is at the halfway point, and 1 is at the end (complete). */
+        totalProgress(): number;
+        totalProgress(value: number, suppressEvents?: boolean): Animation;
+
+        /** Gets or sets the position of the playhead according to the totalDuration which includes any repeats and repeatDelays (only available in TweenMax and TimelineMax). */
+        totalTime(): number;
+        totalTime(time: number, suppressEvents?: boolean): Animation;
```
Comment (DanielRosenwasser) [inline gsap/Animation.d.ts]: Can you collapse each of these into one overload with optional arguments?

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

**Comment by DanielRosenwasser** [comment]
@vilic @codebelt care to look things over?

File: gsap/Animation.d.ts (hunk 2 above)
Comment (ProbablePrime) [inline gsap/Animation.d.ts]: @DanielRosenwasser Unless i'm misunderstanding that alters the return type into a manner that i find un-intuitive.
`totalTime(time?: number, suppressEvents?: boolean): Animation | boolean;` 

Just looks confusing. Is this what you meant?

Comment (bowdenk7) [inline gsap/Animation.d.ts]: @DanielRosenwasser ping

Comment (minestarks) [inline gsap/Animation.d.ts]: @DanielRosenwasser this looks fine to me as is. Will merge unless you object.

**Pull request merged**